bootstrap-salt.sh

```diff
@@ -3627,7 +3632,13 @@ install_centos_stable_deps() {
         yum -y update || return 1
     fi
 
-    if [ $_DISABLE_REPOS -eq $BS_FALSE ]; then
+    if [ "$_DISABLE_REPOS" -eq "$BS_FALSE" ] || ( [ -n "$_PY_EXE" ] && [ "$_PY_MAJOR_VERSION" -eq 3 ] ); then
+
+        if [ "$_DISABLE_REPOS" -eq "$BS_TRUE" ]; then
+            echowarn "Detected -r or -R option while installing Salt packages for Python 3."
+            echowarn "Python 3 packages for Salt require the EPEL repository to be installed."
+        fi
```

Comment:
Here I think `-r` or `-R` options would not do what they supposed to do. Because whenever we detect `python3`, the script will forcibly configure the repositories.
I suggest to simplify the logic: below at line 3638 (`if [ "$_CUSTOM_REPO_URL" != "null" ]; then`) which the `-R` option triggers, add inner test for Py3 and add EPEL (+ warning message) on success.
This would be straight forward and more readable.

Reply:
So really what I was trying to do here is rather than fail when the `-r/-R` options are used in tandem with `-x`, just warn the user and keep going.

But maybe a better approach is to exit with a `warnerror`. Because the Python3 packages on repo.saltstack.com for RHEL require EPEL to be enabled. So until that is no longer the case, it's probably best to check for this and fail with a warning (documented) that the arguments are incompatible.